mpmath/matrices/linalg.py

```diff
@@ -787,3 +839,58 @@ def qr(ctx, A, mode = 'full', edps = 10):
         # ------------------
         # END OF FUNCTION QR
         # ------------------
+
+    def rank(ctx, A, iszerofunc=None):
+        """
+            Calculate the rank of a matrix,
+            i.e. the number of linear independent
+            columns (or rows equivalently).
+
+            Rank is computed via singular value decomposition
+            by counting the number of non-zero singular values.
+
+            **Examples**
+
+            Rank of identity is same as its dimension.
+
+            >>> from mpmath import *
+            >>> A = eye(3)
+            >>> rank(A)
+            3
+
+            But in general a matrix has rank less or equal of its dimension.
+
+            >>> A = matrix([[2, 6, 4],[3, 8, 6],[1, 1, 2]])
+            >>> rank(A)
+            2
+
+            The rank is given by the number of non zero lines in an equivalent triangular matrix.
+
+            >>> R = matrix([[1, 3, 2],[0, 1, 0],[0, 0, 0]])
+            >>> rank(A)
+            2
+
+            The rank is zero if and only if the matrix is zero.
+
+            >>> A = zeros(3)
+            >>> rank(A)
+            0
+
+            The matrix has full rank if and only ist is equivalent to identity,
+
+            >>> A = matrix([[1, 0, 1],[-2, 1, -2],[-4, 1, -5]])
+            >>> rank(A)
+            3
+
+            i.e. has an inverse matrix.
+
+            >>> B = matrix([[3, -1, 1],[2, 1, 0],[-2, 1, -1]])
+            >>> A*B == eye(3)
+            True
+            >>> rank(B)
+            3
+
+        """
+        if iszerofunc is None:
```

Comment:
Only iszerofunc=None was tested, see https://app.codecov.io/gh/mpmath/mpmath/pull/610.  Could you add missing tests?

Reply:
removed `iszerofunc` as `mpmath` handles precision on its own.